packages/venia-concept/src/components/ProductFullDetail/ProductFullDetail.js

```diff
@@ -135,11 +136,45 @@ class ProductFullDetail extends Component {
         );
     }
 
+    get mediaGalleryEntries() {
+        const { props, state } = this;
+        const { product } = props;
+        const { optionCodes, optionSelections } = state;
+        const {
+            configurable_options,
+            media_gallery_entries,
+            variants
+        } = product;
+        const isConfigurable = Array.isArray(configurable_options);
+
+        if (
+            !isConfigurable ||
+            (isConfigurable && optionSelections.size === 0)
+        ) {
+            return media_gallery_entries;
+        }
+
+        const item = findMatchingVariant({
+            optionCodes,
+            optionSelections,
+            variants
+        });
+
+        return item.product.media_gallery_entries;
```

Comment:
If `findMatchingVariant` can return undefined this is unsafe.

Reply:
Resolved by https://github.com/magento-research/pwa-studio/pull/680/commits/ea3fd9490692d2f9298f5a03e7cc05b4f2ad9c03.